api-lib/handlerHelpers.ts

```diff
@@ -0,0 +1,18 @@
+import { VercelRequest } from '@vercel/node';
+import { z } from 'zod';
+
+import {
+  InputSchema,
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+} from '../src/lib/zod';
+
+export function getPropsWithUserSession<T extends z.ZodRawShape>(
```

Comment:
@topocount @crabsinger what do you think of this attempt to slightly reduce boilerplate? i tried more extensive modifications at first but was stymied by TypeScript.

Reply:
Yeah, this looks like a nice step in the right direction. I like to co-location and it feeling a bit more like the handler does the business logic in that one place of a) validate input and b) perform admin graphql actions.